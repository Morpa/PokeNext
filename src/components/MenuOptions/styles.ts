import styled, { css, DefaultTheme } from 'styled-components'

const linkModifiers = {
  default: (theme: DefaultTheme) => css`
    color: ${theme.colors.white};
  `,
  active: (theme: DefaultTheme) => css`
    color: ${theme.colors.yellow};

    &:after {
      content: '';
      width: 8rem;
      height: 0.1rem;
      background: ${theme.colors.yellow};
      position: absolute;
      left: 50%;
      transform: translateX(-50%);
      bottom: 0;
    }
  `
}

type linkProps = {
  isActive?: boolean
}

export const Option = styled.a<linkProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    flex-direction: column;
    position: relative;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    font-size: ${theme.font.sizes.large};
    line-height: ${theme.spacings.small};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small}
      ${theme.spacings.xxsmall} ${theme.spacings.small};

    strong {
      margin-top: ${theme.spacings.xxsmall};
      font-weight: ${theme.font.bold};
    }

    > svg {
      width: 2.4rem;
    }

    &:hover {
      color: ${theme.colors.yellow};
      cursor: pointer;
    }

    ${!isActive && linkModifiers.default(theme)};
    ${isActive && linkModifiers.active(theme)};
  `}
`
