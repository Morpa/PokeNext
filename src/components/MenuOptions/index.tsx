import { useRouter } from 'next/router'
import Link from 'next/link'

import * as S from './styles'

export type MenuOptionsProps = {
  link: string
  linkName: string
  activeLink?: string
}

const MenuOptions = ({ link, linkName }: MenuOptionsProps) => {
  const router = useRouter()

  return (
    <Link href={link} passHref prefetch={false}>
      <S.Option isActive={link === router.asPath}>
        <strong>{linkName}</strong>
      </S.Option>
    </Link>
  )
}

export default MenuOptions
