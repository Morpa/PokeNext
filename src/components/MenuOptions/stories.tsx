import { Story, Meta } from '@storybook/react/types-6-0'

import MenuOptions, { MenuOptionsProps } from '.'

export default {
  title: 'MenuOptions',
  component: MenuOptions,
  args: {
    link: '/link',
    linkName: 'My link',
    activeClassName: 'active'
  },
  argTypes: {
    icon: {
      type: ''
    }
  }
} as Meta

export const Default: Story<MenuOptionsProps> = (args) => (
  <MenuOptions {...args} />
)
