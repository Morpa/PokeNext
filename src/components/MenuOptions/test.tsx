import { screen } from '@testing-library/react'

import MenuOptions from '.'
import { renderWithTheme } from 'utils/tests/helpers'
import theme from 'styles/theme'

const props = {
  link: '/link',
  linkName: 'My link'
}

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({ asPath: '/link' }))
}))

describe('<MenuOptions />', () => {
  it('should render options correctly', () => {
    const { container } = renderWithTheme(<MenuOptions {...props} />)

    expect(screen.getByText(/my link/i)).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /my link/i })).toHaveAttribute(
      'href',
      '/link'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render  when  is active', () => {
    renderWithTheme(<MenuOptions {...props} activeLink="/link" />)

    expect(screen.getByRole('link', { name: /my link/i })).toHaveStyle({
      color: theme.colors.yellow
    })
  })
})
