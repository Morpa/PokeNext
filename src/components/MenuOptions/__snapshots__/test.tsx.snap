// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuOptions /> should render options correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  line-height: 2.4rem;
  padding: 1.6rem 2.4rem 0.8rem 2.4rem;
  color: #FF9F1C;
}

.c0 strong {
  margin-top: 0.8rem;
  font-weight: 700;
}

.c0 > svg {
  width: 2.4rem;
}

.c0:hover {
  color: #FF9F1C;
  cursor: pointer;
}

.c0:after {
  content: '';
  width: 8rem;
  height: 0.1rem;
  background: #FF9F1C;
  position: absolute;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  bottom: 0;
}

<a
  class="c0"
  href="/link"
>
  <strong>
    My link
  </strong>
</a>
`
